criu/include/criu-plugin.h

```diff
@@ -69,8 +69,8 @@ DECLARE_PLUGIN_HOOK_ARGS(CR_PLUGIN_HOOK__DUMP_EXT_MOUNT, char *mountpoint, int i
 DECLARE_PLUGIN_HOOK_ARGS(CR_PLUGIN_HOOK__RESTORE_EXT_MOUNT, int id, char *mountpoint, char *old_root, int *is_file);
 DECLARE_PLUGIN_HOOK_ARGS(CR_PLUGIN_HOOK__DUMP_EXT_LINK, int index, int type, char *kind);
 DECLARE_PLUGIN_HOOK_ARGS(CR_PLUGIN_HOOK__HANDLE_DEVICE_VMA, int fd, const struct stat *stat);
```

Comment:
is it better to pass file path of fd?

Reply:
Hi @time-river,
I have been working with @rajbhar for this plugin. It is great to see more contributors to this project.

> typedef int(cr_plugin_handle_device_vma_t)(const char *path, const struct stat *st_buf, struct vma_area *vma);

- We are ok with using path instead of fd. 
- We do not need the vma parameter, so this would not affect us. When designing our hooks, we tried to avoid exposing CRIU internal structures to the plugins as this could limit the ability to modify these structures in the future and risk ABI breakage. We would be ok either way, @avagin @Snorch please let us know your views on it.

> typedef int(cr_plugin_update_vma_map_t)(const char *path, struct vma_area *vma);
- Similar concern as above regarding passing vma to the plugin. Functionally, passing in the vma would work for us.

> sometimes, the vma is assigned to the specific fd
- We have one additional patch in this series that come after this one:
https://github.com/RadeonOpenCompute/criu/commit/41c7c429746e22b1cf04332239d2c1fff331907b
The plugin may return a fd that will later be used by CRIU when calling mmap. Would this fit your needs?

Please also note that this commit has already been pulled into the criu-dev branch, so you may want to create a new PR for the requested changes.